test/no_only.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var tap = require('tap');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var concat = require('concat-stream');
+
+var stripFullStack = require('./common').stripFullStack;
+
+var tapeBin = path.join(__dirname, '../bin/tape');
+
+var expectedFailure = (/^0\.10\.\d+$/).test(process.versions.node);
+
+tap.test(
+    'Should throw error when --no-only is passed via cli and there is a .only test',
+    { todo: expectedFailure ? 'Fails on these node versions' : false },
+    function (tt) {
+        tt.plan(2);
+
+        var testStderr = function (rows) {
+            tt.ok((/Error: `only` tests are prohibited\n/).test(stripFullStack(rows.toString('utf8')).join('\n')));
+        };
+
+        var ps = spawn(tapeBin, ['--no-only', '**/*.js'], { cwd: path.join(__dirname, 'no_only') });
+        ps.stderr.pipe(concat(testStderr));
+        ps.on('exit', function (code) {
+            tt.equal(code, 1);
+        });
+    }
+);
+
+tap.test(
+    'Should throw error when NODE_TAPE_NO_ONLY_TEST is passed via envs and there is an .only test',
+    { todo: expectedFailure ? 'Fails on these node versions' : false },
+    function (tt) {
+        tt.plan(2);
+
+        var testStderr = function (rows) {
+            tt.ok((/Error: `only` tests are prohibited\n/).test(stripFullStack(rows.toString('utf8')).join('\n')));
+        };
+        var ps = spawn(tapeBin, ['**/*.js'], {
+            cwd: path.join(__dirname, 'no_only'),
+            env: { PATH: process.env.PATH, NODE_TAPE_NO_ONLY_TEST: false }
```

Comment:
for this, the test case was initially meant to test out behaviour when `NODE_TAPE_NO_ONLY_TEST` is passed as `true` and there is an existing `only` test. A bit confused about the change to `false`?

Reply:
oh oops! Just a bad copy-paste. Will fix.

Reply:
o okay cool, but anyway, was wondering why the related tests passed anyway, tried locally and realized the env is evaluated as a string in the code so always evaluates to `true`